distribution/integration/tests-integration/src/test/java/org/wso2/carbon/integration/tests/carbontools/CarbonServerBasicOperationTestCase.java

```diff
@@ -0,0 +1,219 @@
+        }
+
+        boolean isServerDown = CarbonCommandToolsUtil.isServerDown(context, portOffset);
+        assertTrue(isServerDown, "Shutting down the server failed");
+
+        boolean isServerUp = CarbonCommandToolsUtil.isServerStartedUp(context, portOffset);
+
+        assertTrue(isServerUp, "Unsuccessful login");
+    }
+
+    @Test(groups = {"wso2.as"}, description = "Server stop test",
+            dependsOnMethods = {"testRestartCommand"})
+    public void testStopCommand() throws Exception {
+        String[] cmdArray;
+        boolean startupStatus = false;
+        try {
+            if (CarbonCommandToolsUtil.isCurrentOSWindows()) {
+                throw new SkipException("Feature --stop is not available for windows");
+            } else {
+                cmdArray = new String[]{"sh", "wso2server.sh", "--stop", "-DportOffset=" + portOffset};
+                processStop = CarbonCommandToolsUtil.runScript(carbonHome + "/bin", cmdArray);
+            }
+            startupStatus = CarbonCommandToolsUtil.isServerDown(context, portOffset);
+        } finally {
+            if (processStop != null) {
+                processStop.destroy();
+            }
+        }
+        assertTrue(startupStatus, "Unsuccessful login");
+    }
+
+    @Test(groups = {"wso2.as"}, description = "Server stop test")
+    public void testDumpCommandOnWindows() throws Exception {
+        Process processDump = null;
+        String carbonHome = System.getProperty(ServerConstants.CARBON_HOME);
+        try {
+            if (CarbonCommandToolsUtil.isCurrentOSWindows()) {
+                RuntimeMXBean runtimeMXBean = ManagementFactory.getRuntimeMXBean();
+                Field jvmField = runtimeMXBean.getClass().getDeclaredField("jvm");
+                jvmField.setAccessible(true);
+                VMManagement vmManagement = (VMManagement) jvmField.get(runtimeMXBean);
+                Method getProcessIdMethod = vmManagement.getClass().getDeclaredMethod("getProcessId");
+                getProcessIdMethod.setAccessible(true);
+                Integer processId = (Integer) getProcessIdMethod.invoke(vmManagement);
+                String[] cmdArray = new String[]
+                        {"cmd.exe", "/c", "carbondump.bat", "-carbonHome"
+                                , carbonHome, "-pid", Integer.toString(processId)};
+                processDump = CarbonCommandToolsUtil.runScript(carbonHome + "/bin", cmdArray);
+                assertTrue(isFoundDumpFile(carbonHome), "Couldn't find the dump file");
+            } else {
+                throw new SkipException(" This test method is only for windows");
+            }
+        } finally {
+            if (processDump != null) {
+                processDump.destroy();
+            }
+        }
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void serverShutDown() throws Exception {
+        Runtime.getRuntime().addShutdownHook(new Thread() {
+            public void run() {
+                try {
```

Comment:
1. Change Exception to exact exception you get
2. throw the exception after log or don't catch it here